great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -966,6 +956,25 @@ def _build_selectable_from_batch_spec(self, batch_spec) -> Union[Selectable, str
                         .where(split_clause)
                         .limit(batch_spec["sampling_kwargs"]["n"])
                     )
+            elif batch_spec["sampling_method"] == "_sample_using_random":
+                num_rows: int = self.engine.execute(
+                    sa.select([sa.func.count()])
+                    .select_from(
+                        sa.table(table_name, schema=batch_spec.get("schema_name", None))
+                    )
+                    .where(split_clause)
+                ).one()[0]
+                p: Optional[float] = batch_spec["sampling_kwargs"]["p"] or 1.0
+                sample_size: int = round(p * num_rows)
+                return (
+                    sa.select("*")
+                    .select_from(
+                        sa.table(table_name, schema=batch_spec.get("schema_name", None))
+                    )
+                    .where(split_clause)
+                    .order_by(sa.func.random())
```

Comment:
❤️

Reply:
clarification: 

sa.func.random() is a roll of a dice. Only when combined with `order_by()` can filter rows randomly